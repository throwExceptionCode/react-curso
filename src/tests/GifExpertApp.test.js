import React from 'react';
import { shallow } from "enzyme";
import { GifExpertApp } from '../gifExpertApp';

describe('Pruebas de Componente GifExpert', () => {
    test('should be match with snapshot', () => {
        const wrapper = shallow(<GifExpertApp  />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should be show a list of category', () => {
        const categories = ['dog','dig'];
        const wrapper = shallow(<GifExpertApp defaultCategory={ categories } />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);    
    });

    
    
})
