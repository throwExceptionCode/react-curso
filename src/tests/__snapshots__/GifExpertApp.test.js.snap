// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de Componente GifExpert should be match with snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
  <hr />
</Fragment>
`;

exports[`Pruebas de Componente GifExpert should be show a list of category 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="dog"
      key="dog"
    />
    <GifGrid
      category="dig"
      key="dig"
    />
  </ol>
  <hr />
</Fragment>
`;
